presto-main/src/main/java/io/prestosql/split/SplitManager.java

```diff
@@ -125,7 +125,7 @@ public SplitSource getSplits(Session session, TableHandle table, SplitScheduling
         if (minScheduleSplitBatchSize > 1) {
             splitSource = new BufferingSplitSource(splitSource, minScheduleSplitBatchSize);
         }
-        if (SystemSessionProperties.isSnapshotEnabled(session)) {
+        if (SystemSessionProperties.isSnapshotEnabled(session) && !SystemSessionProperties.isSnapshotCaptureDisabled(session)) {
```

Comment:
Recovery enabled and snapshot enabled.. these 2 only should be configs... IMO

Reply:
This check is to avoid sending marker split when disabling snapshot capture (newly added config) is enabled. 
There is no recovery with snapshot flag disabled, So did not introduce new flag for recovery again.